packages/app/src/cli/services/dev/graphiql/server.ts

```diff
@@ -87,87 +53,141 @@ export function setupGraphiQLServer({
       next()
     })
 
-  let session: Session | undefined
-
-  async function beginAuth(req: express.Request, res: express.Response) {
-    stdout.write('Initiating OAuth flow...')
-    await shopify.auth.begin({
-      shop: shopify.utils.sanitizeShop(storeFqdn, true)!,
-      callbackPath: `/${urlNamespaces.devTools}/graphiql/auth/callback`,
-      isOnline: false,
-      rawRequest: req,
-      rawResponse: res,
-    })
+  let _token: string | undefined
+  async function token(): Promise<string> {
+    if (!_token) {
+      // eslint-disable-next-line require-atomic-updates
+      _token = await refreshToken()
+    }
+    return _token
   }
 
-  async function isAuthorized(): Promise<boolean> {
-    if (!session) return false
-    const client = new shopify.clients.Graphql({session, apiVersion: LATEST_API_VERSION})
+  async function refreshToken(): Promise<string> {
     try {
-      await client.query({data: '{ shop { id } }'})
-      return true
-      // eslint-disable-next-line no-catch-all/no-catch-all
-    } catch (error) {
-      return (error as HttpResponseError)?.response?.code !== 401
+      outputDebug('refreshing token', stdout)
+      _token = undefined
+      const queryString = queryStringEncode({
+        client_id: apiKey,
+        client_secret: apiSecret,
+        grant_type: 'client_credentials',
+      })
+      const tokenResponse = await fetch(`https://${storeFqdn}/admin/oauth/access_token?${queryString}`, {
```

Comment:
are there any plans to have this in the `@shopify/shopify-api` library?

Reply:
I agree it's good to not have the full shopify-api dependency. Maybe worth moving it to CLI Kit though to start with?